### DISCUSSION TITLE
Touch screen functions

### DISCUSSION THREAD

**Discussion by steve8x8** [OpenNeptune3D/OpenNept4une#24] [Q&A]
I know this is Work In Progress (and I appreciate that it's being worked on), but I have a few questions after I had found the time to upgrade using OpenNep.sh ...
1. There's a Temperature submenu in Settings - with no tempratures assigned. If I added some, what exactly would those be used for?
2. I clicked Level and got a screen I couldn't leave anymore - how would I reset the display in such a case?
Thanks for working on this, and thanks in advance for any hints to fix my ignorance!

**Comment by halfmanbear** [comment]
Are you reffering to the temps within different filament categories . These are just preset temps for different print materials.

There are many bugs and pages that haven't been programmed for the screen as of yet, including the levelling page. 

I might disable any page nav to broken pages untill they are developed also we need to put a function to monitor the moonraker connection for auto reconnect if the firmware / klipper restarts.

Also need a serial detect when the screen is unplugged / reconnected so it lands on the main page

**Comment by steve8x8** [comment]
While following WiP as much as I can, I'm now wondering whether screen functions would depend on the actual `.tft` file installed (before changing to OpenNep*) - does that matter at all? (Apparently not as it's mentioned nowhere?)

**Reply by halfmanbear** [reply]
Seemingly not, although we are using this display .tft for our development currently

https://github.com/halfmanbear/OpenNept4une/blob/main/display/dev-resources/ui_1.2.11_231012.tft

**Reply by steve8x8** [reply]
... which is newer than mine (I think 1.1.8 since mine was built in August and never updated with one of Elegoo's newer releases) but I'll be ready for surprises

**Reply by phillipthelen** [reply]
@steve8x8 updating the touch screen can easily be done with the micro sd card that came with the printer. The 1.2.11 and 1.2.12 versions are the ones we work with, but chances are your older version will also work just the same. If you are hesitant about updating the display, you can just try it out. Worst case you would run into issues where things do not display correctly (text fields would be blank or show the incorrectly default data) or a button might not work. But there isn't anything "bad" that would happen due to this. And when you do run into those issues, you can always still update then.
